make/common/JavaCompilation.gmk

```diff
@@ -442,10 +468,18 @@ define SetupJavaCompilationBody
       $1_EXTRA_DEPS := $$(BUILDTOOLS_OUTPUTDIR)/depend/_the.COMPILE_DEPEND_batch
     endif
 
+    addjpp:
+      ifeq ($1, java.base)
+		$(MKDIR) -p $(SUPPORT_OUTPUTDIR)/overlay/src/java.base/sun/nio/cs
+		cp $(SUPPORT_OUTPUTDIR)/gensrc/java.base/sun/nio/cs/StandardCharsets.java $(SUPPORT_OUTPUTDIR)/overlay/src/java.base/sun/nio/cs
+		$$(call RunJPP, JAVA$(VERSION_FEATURE), $(SUPPORT_OUTPUTDIR)/overlay, /overlay-result)
+		cp $(SUPPORT_OUTPUTDIR)/j9jcl/overlay-result/java.base/sun/nio/cs/StandardCharsets.java $(SUPPORT_OUTPUTDIR)/gensrc/java.base/sun/nio/cs
+      endif
+
     # Create a file with all sources, to pass to javac in an @file.
     # $$($1_VARDEPS_FILE) is used as dependency to track changes in set of
     # list of files.
-    $$($1_FILELIST): $$($1_SRCS) $$($1_VARDEPS_FILE)
+    $$($1_FILELIST): $$($1_SRCS) $$($1_VARDEPS_FILE) addjpp
```

Comment:
Making all java compiles unconditional is not a reasonable solution.

I suggest we need a rule that depends on the target defined by `GensrcCharsetMapping.gmk` which runs the preprocessor on the output of that step.

Reply:
Moved JPP call into `GensrcCharsetMapping.gmk`.